libpromises/eval_context.c

```diff
@@ -450,7 +455,10 @@ bool IsDefinedClass(const EvalContext *ctx, const char *context)
         return true;
     }
 
-    res = ParseExpression(context, 0, strlen(context));
+    Buffer *condensed = BufferNewFrom(context, strlen(context));
+    BufferRewrite(condensed, &ClassCharIsWhitespace, true);
+    res = ParseExpression(BufferData(condensed), 0, BufferSize(condensed));
+    BufferDestroy(condensed);
```

Comment:
Should we do the "no operator in between" check here as well, in addition to (or perhaps instead of) above? Then the check will work for both quoted and non-quoted strings, unless I'm mistaken.